hudi-common/src/main/java/org/apache/hudi/common/util/ClusteringUtils.java

```diff
@@ -68,21 +69,27 @@
         .filter(Option::isPresent).map(Option::get);
   }
 
-  public static Option<Pair<HoodieInstant, HoodieClusteringPlan>> getClusteringPlan(HoodieTableMetaClient metaClient, HoodieInstant requestedReplaceInstant) {
+  public static Option<Pair<HoodieInstant, HoodieClusteringPlan>> getClusteringPlan(HoodieTableMetaClient metaClient, HoodieInstant pendingReplaceInstant) {
     try {
-      Option<byte[]> content = metaClient.getActiveTimeline().getInstantDetails(requestedReplaceInstant);
+      final HoodieInstant requestedInstant;
+      if (!pendingReplaceInstant.isRequested()) {
```

Comment:
can we add some annotation such as "do not have content in inflight instant"

Reply:
@satishkotha Can you add a comment here as to why this is needed ?

Reply:
Added comment. PTAL